clang-tools-extra/test/clang-tidy/checkers/bsl-destructor-access-specifier.cpp

```diff
@@ -0,0 +1,57 @@
+// RUN: %check_clang_tidy %s bsl-destructor-access-specifier %t
+
+class A
```

Comment:
These test cases looks good. I would add some with template classes too and make sure it is being instantiated in a function somewhere. E.g.

```cpp
template <typename T>
class X {
private:
    T t;
    T get_t() { return t; }
public:
    ~X() = default; // non-compliant, class is not final
};

int foo()
{
    X<int> x;
   return x.get_t();
}
```